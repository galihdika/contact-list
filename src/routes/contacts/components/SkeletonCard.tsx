import { Card, Skeleton } from "antd";

const SkeletonCard = () => {
  return (
    <Card
      data-testid="skeleton-card"
      hoverable
      cover={<div className="card-cover-skeleton"><Skeleton.Image active /></div>}
      actions={[
        <Skeleton.Button active size="small"/>,
        <Skeleton.Button active size="small"/>,
        <Skeleton.Button active size="small"/>,
      ]}
    >
      <Card.Meta 
        title={<Skeleton active />}
      />
    </Card>
  )
}

export default SkeletonCard;