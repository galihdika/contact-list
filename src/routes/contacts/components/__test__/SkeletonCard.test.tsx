import { render } from "@testing-library/react";

import SkeletonCard from "../SkeletonCard";

describe('test SkeletonCard', () => {
  test('should match snapshot', () => {
    const { container } = render(<SkeletonCard/>);

    expect(container).toMatchSnapshot();
  })
})