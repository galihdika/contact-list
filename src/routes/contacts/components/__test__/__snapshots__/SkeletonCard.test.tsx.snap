// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SkeletonCard should match snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-1wazalj"
  >
    <div
      class="ant-card-cover"
    >
      <div
        class="card-cover-skeleton"
      >
        <div
          class="ant-skeleton ant-skeleton-element ant-skeleton-active css-dev-only-do-not-override-1wazalj"
        >
          <div
            class="ant-skeleton-image"
          >
            <svg
              class="ant-skeleton-image-svg"
              viewBox="0 0 1098 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="ant-skeleton-image-path"
                d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-card-meta"
      >
        <div
          class="ant-card-meta-detail"
        >
          <div
            class="ant-card-meta-title"
          >
            <div
              class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-1wazalj"
            >
              <div
                class="ant-skeleton-content"
              >
                <h3
                  class="ant-skeleton-title"
                  style="width: 38%;"
                />
                <ul
                  class="ant-skeleton-paragraph"
                >
                  <li />
                  <li />
                  <li
                    style="width: 61%;"
                  />
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <ul
      class="ant-card-actions"
    >
      <li
        style="width: 33.333333333333336%;"
      >
        <span>
          <div
            class="ant-skeleton ant-skeleton-element ant-skeleton-active css-dev-only-do-not-override-1wazalj"
          >
            <span
              class="ant-skeleton-button ant-skeleton-button-sm"
            />
          </div>
        </span>
      </li>
      <li
        style="width: 33.333333333333336%;"
      >
        <span>
          <div
            class="ant-skeleton ant-skeleton-element ant-skeleton-active css-dev-only-do-not-override-1wazalj"
          >
            <span
              class="ant-skeleton-button ant-skeleton-button-sm"
            />
          </div>
        </span>
      </li>
      <li
        style="width: 33.333333333333336%;"
      >
        <span>
          <div
            class="ant-skeleton ant-skeleton-element ant-skeleton-active css-dev-only-do-not-override-1wazalj"
          >
            <span
              class="ant-skeleton-button ant-skeleton-button-sm"
            />
          </div>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
